akka-persistence-query/src/test/scala/akka/persistence/query/journal/leveldb/EventsByTagSpec.scala

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.query.journal.leveldb
+
+import scala.concurrent.duration._
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.persistence.query.PersistenceQuery
+import akka.persistence.query.RefreshInterval
+import akka.stream.ActorMaterializer
+import akka.stream.testkit.scaladsl.TestSink
+import akka.testkit.ImplicitSender
+import akka.testkit.TestKit
+import akka.persistence.query.NoRefresh
+import akka.testkit.AkkaSpec
+import akka.persistence.query.EventsByTag
+import akka.persistence.journal.leveldb.Tagged
+import akka.persistence.journal.EventSeq
+import akka.persistence.journal.EventAdapter
+import akka.persistence.query.EventEnvelope
+
+object EventsByTagSpec {
+  val config = """
+    akka.loglevel = INFO
+    akka.persistence.journal.plugin = "akka.persistence.journal.leveldb"
+    akka.persistence.journal.leveldb {
+      dir = "target/journal-EventsByTagSpec"
+      event-adapters {
+        color-tagger  = akka.persistence.query.journal.leveldb.ColorTagger
+      }
+      event-adapter-bindings = {
+        "java.lang.String" = color-tagger
+      }
+    }
+    """
+
+}
+
+class ColorTagger extends EventAdapter {
+  val colors = Set("green", "black", "blue")
+  override def toJournal(event: Any): Any = event match {
+    case s: String ⇒
+      var tags = colors.foldLeft(Set.empty[String])((acc, c) ⇒ if (s.contains(c)) acc + c else acc)
+      if (tags.isEmpty) event
+      else Tagged(event, tags)
+    case _ ⇒ event
+  }
+
+  override def fromJournal(event: Any, manifest: String): EventSeq = EventSeq.single(event)
```

Comment:
// PR unrelated: btw each time I see this it makes me want to go back and try to make `EventSeq` a value class somehow ;-)